FoodOrderingApp-service/src/main/java/com/upgrad/FoodOrderingApp/service/businness/ItemService.java

```diff
@@ -1,20 +1,78 @@
 package com.upgrad.FoodOrderingApp.service.businness;
 
+import com.upgrad.FoodOrderingApp.service.common.UtilityProvider;
 import com.upgrad.FoodOrderingApp.service.dao.ItemDao;
+import com.upgrad.FoodOrderingApp.service.dao.OrderDao;
+import com.upgrad.FoodOrderingApp.service.dao.OrderItemDao;
 import com.upgrad.FoodOrderingApp.service.entity.ItemEntity;
+import com.upgrad.FoodOrderingApp.service.entity.OrderItemEntity;
+import com.upgrad.FoodOrderingApp.service.entity.OrdersEntity;
 import com.upgrad.FoodOrderingApp.service.entity.RestaurantEntity;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.HashMap;
+import java.util.LinkedList;
 import java.util.List;
+import java.util.Map;
 
 @Service
 public class ItemService {
 
     @Autowired
     private ItemDao itemDao;
 
-    public List<ItemEntity> getItemsByPopularity(final RestaurantEntity restaurantEntity) {
-        return itemDao.getItemsByPopularity(restaurantEntity.getId());
+    @Autowired
+    private OrderDao orderDao;
+
+    @Autowired
+    private OrderItemDao orderItemDao;
+
+    @Autowired
+    private UtilityProvider utilityProvider;
+
+    /* This method is to get Items By Popularity and returns list of ItemEntity it takes restaurantEntity as input.
+     */
+    public List<ItemEntity> getItemsByPopularity(RestaurantEntity restaurantEntity) {
```

Comment:
Any reason for overriding the implementation I wrote?

Reply:
no reason even if u make it private still output came the same